### PR TITLE
Problem: push0 opcode is not tested

### PR DESCRIPTION
**Pull request by yihuang**
- shanghai hardfork introduces push0 opcode, can be enabled in solc

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

## Summary by CodeRabbit

- **Documentation**
  - Updated the Solidity compiler version in documentation to reflect the latest version used in smart contracts.

- **New Features**
  - Introduced support for hard-fork style upgrades in integration tests.
  - Enabled the `push0` opcode in integration tests for enhanced smart contract capabilities.

- **Bug Fixes**
  - Adjusted gas usage expectations in integration tests to align with the latest contract behavior.

- **Tests**
  - Improved test configurations by adding new markers for unmarked and gas-related tests.
  - Enhanced test cases to assert correct gas usage after contract interactions.
  - Included additional test steps to wait for new blocks before transactions, ensuring more reliable test execution.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## UNRELEASED
 
 - [#1258](https://github.com/crypto-org-chain/cronos/pull/1258) Support hard-fork style upgrades.
+- [#]() Enable push0 opcode in integration test
```
Comment (yihuang) [inline CHANGELOG.md:6]:
```suggestion
- [#1273](https://github.com/crypto-org-chain/cronos/pull/1273) Enable push0 opcode in integration test
```

**Review by mmsqe — APPROVED** [review]

File: integration_tests/test_upgrade.py [hunk 2]
```diff
@@ -123,7 +123,21 @@ def exec(c, tmp_path_factory, testnet=True):
...
+        assert 'invalid opcode: PUSH0' in str(e_info.value)
+
     contract = deploy_contract(w3, CONTRACTS["TestERC20A"])
+
+    tx = contract.functions.transfer(ADDRS["community"], 10).transact(
+        {"from": ADDRS["validator"]}
+    )
+    tx_receipt = w3.eth.wait_for_transaction_receipt(tx)
+    print(tx_receipt)
+
```
Comment (yihuang) [inline integration_tests/test_upgrade.py:140]:
```suggestion
```

File: integration_tests/test_basic.py [hunk 3]
```diff
@@ -63,6 +64,7 @@ def test_events(cluster, suspend_capture):
         w3,
         CONTRACTS["TestERC20A"],
         key=KEYS["validator"],
+        exp_gas_used=633810,
```
Comment (mmsqe) [inline integration_tests/test_basic.py:67]: might use more gas in [older](https://github.com/crypto-org-chain/cronos/blob/18462e2960ad649e766a018b361cd2e63f4b7045/integration_tests/contracts/contracts/TestERC20A.sol#L1) version

Comment (yihuang) [inline integration_tests/test_basic.py:67]:
```suggestion
        exp_gas_used=641641,
```

Comment (yihuang) [reply integration_tests/test_basic.py]: you are right

**Review by mmsqe — APPROVED** [review]

**Pull request merged**